tests/zquantum/qcbm/ansatz_utils_test.py

```diff
@@ -76,6 +79,180 @@ def test_get_entangling_layer_line_topology(self):
         self.assertEqual(ent_layer.gates[2].qubits[0].index, 2)
         self.assertEqual(ent_layer.gates[2].qubits[1].index, 3)
 
+    def test_get_entangling_layer_star_topology(self):
+        # Given
+        n_qubits = 4
+        single_qubit_gate = "Rx"
+        static_entangler = "XX"
+        topology = "star"
+        center_qubit = 1
+        params = np.asarray([0, 0, 0])
+
+        # When
+        ent_layer = get_entangling_layer_star_topology(
+            params, n_qubits, static_entangler, center_qubit
+        )
+
+        # Then
+        for gate in ent_layer.gates:
+            self.assertTrue(gate.name, "XX")
+
+        # XX on 0, 1
+        self.assertEqual(ent_layer.gates[0].qubits[0].index, 0)
+        self.assertEqual(ent_layer.gates[0].qubits[1].index, 1)
+        # XX on 1, 2
+        self.assertEqual(ent_layer.gates[1].qubits[0].index, 1)
+        self.assertEqual(ent_layer.gates[1].qubits[1].index, 2)
+        # XX on 1, 3
+        self.assertEqual(ent_layer.gates[2].qubits[0].index, 1)
+        self.assertEqual(ent_layer.gates[2].qubits[1].index, 3)
+
+    def test_get_entangling_layer_graph_topology_matrix1(self):
```

Comment:
You can have test cases for multiple inputs (e.g. matrix or graph) by using `pytest.mark.parameterize` feature.
You can find it's used in many places in our codebase, for example [here](https://github.com/zapatacomputing/z-quantum-core/blob/a54bde2688d7bf2c37538d75f76a02ce27b94b34/tests/zquantum/core/hamiltonian_test.py#L183) .